source/extensions/clusters/logical_dns/logical_dns_cluster.cc

```diff
@@ -101,7 +102,11 @@ void LogicalDnsCluster::startResolve() {
         active_dns_query_ = nullptr;
         ENVOY_LOG(trace, "async DNS resolution complete for {} details {}", dns_address_, details);
 
-        std::chrono::milliseconds final_refresh_rate = dns_refresh_rate_ms_;
+        std::chrono::milliseconds jitter(0);
+        if (dns_jitter_ms_.count() != 0) {
+          jitter = std::chrono::milliseconds(random_.random()) % dns_jitter_ms_;
```

Comment:
I think it may make more sense to add jitter to `final_refresh_rate` only once after line 156, and the `jitter` variable can be discarded.